### ISSUE TITLE
Header can not be modified when datatable is filled from promise or ajax

### ISSUE THREAD

**Issue by pidupuis** [l-lin/angular-datatables#149]
Hi everybody,

I am implementing a search interface and according to the kind of data you want to search for, the datatable will not contain the same set of columns.

First, I want to display the datatable only when the search button is clicked. For this I used a little trick with ng-if I found here: https://github.com/l-lin/angular-datatables/issues/22#issuecomment-60269415.

So my function looks like this:

```javascript
self.doSearch = function() {
    var dtColumns = [];
    for (var i=0; i < self.selectedIdType.header.length; i++) {
        dtColumns.push(DTColumnDefBuilder.newColumnDef(i).withTitle(self.selectedIdType.header[i]));
    }
    self.dtColumnDefs = dtColumns;

    self.dtOptions = DTOptionsBuilder.fromFnPromise(self.selectedIdType.getDataFunc(self.asyncSelected))
        .withPaginationType('full_numbers');

    self.showDT = true;
};
```

with HTML:
```HTML
<table ng-if="Ctrl.showDT" datatable="" dt-options="Ctrl.dtOptions" dt-column-defs="Ctrl.dtColumnDefs" class="row-border hover"></table>
```

It works well for the first search but when I want to search for another kind of data, the datatable is filled with the good result, but the columns are not modified. Therefore, I obtain an error popup each time the first number of column does not fit the new content. And also the header values are not modified.

I also tried to modify self.dtOptions at the last loop iteration but nothing changes...

Why the header is not updated but the rows are? What am I doing wrong?

Thanks in advance for your help!

**Comment by l-lin** [comment]
Fixed on the upcoming version 0.4.0.

**Comment by pidupuis** [comment]
Thanks a lot!

Even my plunker is solved since it uses the dev version of angular-datatables :)
